src/sql/src/plan/statement/ddl.rs

```diff
@@ -1710,6 +1766,82 @@ pub(crate) fn load_generator_ast_to_generator(
                 count_clerk,
             }
         }
+        mz_sql_parser::ast::LoadGenerator::KeyValue => {
+            scx.require_feature_flag(&vars::ENABLE_LOAD_GENERATOR_KEY_VALUE)?;
+            let LoadGeneratorOptionExtracted {
+                keys,
+                snapshot_rounds,
+                transactional_snapshot,
+                value_size,
+                tick_interval,
+                seed,
+                partitions,
+                batch_size,
+                ..
+            } = options.to_vec().try_into()?;
```

Comment:
There is a variable named `extracted` in scope with this conversion and it also exists in the caller of this function in line 1082. Let's convert the options once somewhere in `1082` and pass a `&LoadGeneratorOptionExtracted` in this funciton so that we don't do the conversion over and over.

There is also a call to `ensure_only_valid_options` here that should be hoisted

Reply:
Oh this is because of a bad rebase, will fixup